designs/dedicatedhosts.md

```diff
@@ -0,0 +1,86 @@
+# Dedicated host support
+## Background
+
+As described in [3182](https://github.com/aws/karpenter/issues/3182) AWS provides the 
+ability to launch EC2 instances on [Dedicated Host hardware](https://docs.aws.amazon.com/license-manager/latest/userguide/host-resource-groups.html)
+Dedicated hosts are often used to pay for ec2 usage as capital expenditure (CapEx)
+rather than an operating expense (OpEx). This is often useful for publicly listed
+companies that want to manage their revenue to OpEx ratio.
+
+AWS allows the allocation of ec2 instances to Dedidcated Host Host Resource Groups (HRG)
+through the use of launchtemplates. 
+Detailed documentation for launchtemplates and host resource groups can be found here:
+- https://docs.aws.amazon.com/autoscaling/ec2/userguide/create-launch-template.html#advanced-settings-for-your-launch-template
+- https://docs.aws.amazon.com/license-manager/latest/userguide/host-resource-groups.html
+
+## Solutions
+
+aws-karpenter already supports a set of [LaunchTemplate configuration](https://github.com/aws/karpenter/blob/main/pkg/apis/v1alpha1/awsnodetemplate.go)
+with design document: https://github.com/aws/karpenter/blob/main/designs/aws-launch-templates-v2.md
+
+1. Implement the minimal fields from https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata-placement.html
+specifically `placement.hostResourceGroupArn` and `licenseConfiguration`
+
+this would look like:
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  licenseConfiguration: arn:aws:license-manager:eu-east-1:123456789012:license-configuration:lic-edf7f9e241f5e16f29996c842111f448 # optional, arn of the license configuration
```

Comment:
I looked at the APIs very briefly but I see that you can tag both license configurations and resource groups. I think we may be able to achieve discovery here since both resource groups and license configuration have tags and naming.

Granted, we may have to do some client-side filtering for the license configuration since it looks like you can't filter on tags directly from the server-side.